pgrx/src/pg_catalog.rs

```diff
@@ -0,0 +1,1212 @@
+//LICENSE Portions Copyright 2019-2021 ZomboDB, LLC.
+//LICENSE
+//LICENSE Portions Copyright 2021-2023 Technology Concepts & Design, Inc.
+//LICENSE
+//LICENSE Portions Copyright 2023-2023 PgCentral Foundation, Inc. <contact@pgcentral.org>
+//LICENSE
+//LICENSE All rights reserved.
+//LICENSE
+//LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+use crate as pgrx;
+use pgrx::datum::Array;
+use pgrx::pg_sys::Oid;
+use pgrx::pg_sys::Oid as Regproc;
+use std::ffi::{c_char, CStr};
+
+unsafe trait GetStruct<T> {
+    unsafe fn get_struct(raw: *const T) -> Self;
+}
```

Comment:
Oh, and what's the safety contract of this trait? i.e. what is the promise that `unsafe impl GetStruct` must uphold?

Reply:
Removed `unsafe` in line `17` and added safety for `fn get_struct`.